### ISSUE TITLE
Some radios doesn't buffer fast enough and make Nadeko cancel it

### ISSUE THREAD

**Issue by Nitix** [Kwoth/NadekoBot#601]
Some radios doesn't buffer at time, so Nadeko will stop them even if it receive data.

**Comment by Jinnial** [comment]
For me, it is not just some but all radio streams

**Comment by hackthecode** [comment]
Same here, after update to 0.99.9z
-----
Prebuffering timed out or canceled. Cannot get any data from the stream.
l
Buffering stopped: A task was canceled.
Buffering done.

**Comment by nowherejan** [comment]
I have problems not only with radio streams, but with youtube videos via !!q and !!yq, I have the buffering errors hackthecode mentioned. This was on the y and z builds.

**Comment by Muttsu** [comment]
I have the same problem for most radio (only found 1 that words) and YouTube, but also this happens:
```    
Buffering done.
Buffering successfull
Prebuffering successfully completed in 00:00:01.2039529
l
```
or
```
Prebuffering timed out or canceled. Cannot get any data from the stream.
l
Stream mapping:
  Stream #0.0 -> #0.0
Press ctrl-c to stop encoding
Error while decoding stream #0.0
Buffering stopped: A task was canceled.
Buffering done.
```
The stream just ends without playing anything, and the NadekoBot message is deleted automaticaly.
According to GoogleAPIs there were no errors on their end.

**Comment by Juhni** [comment]
Radios doesn't work for me either. For an example this Finnish radio works on Nadeko:
http://stream.basso.fi:8000/stream

But this does not:
http://audio.raa.fi:8000/radiosputnik.m3u

Both are hosted on Icecast 2.4.2 and they work flawlessly on VLC. The error is the same for many other radios too:
"Prebuffering timed out or canceled. Cannot get any data from the stream. Buffering stopped: A task was canceled. Buffering done."

**Comment by Disinterpreter** [comment]
I agree this problem.

**Comment by fictionalnumber** [comment]
having this same issue

**Comment by IngveBoe** [comment]
me to

**Comment by DarkRsing** [comment]
Any words on this? I used to work perfectly for me

**Comment by ZirconiumHacker** [comment]
Happens to me as well.

**Comment by Nitix** [comment]
After a long time this should be fixed with the PR #755. If some of you want to test it go ahead and report bugs and the PR.

**Comment by Nitix** [comment]
Closed as PR #755 is merged
